In [1]:
from astropy import constants as C
from astropy import units as U
from astropy import table
from math import pi, sqrt
import numpy as np
from scipy.integrate import ode

For this HW we are going to do everything in GeV, so define fundamental constants in these energy units

In [2]:
T0 = 2.7 * U.Kelvin
T0_E = (T0 * C.k_B).to(U.GeV)
T0_E

<Quantity 2.326679743939458e-13 GeV>

In [3]:
H0 = 65 * U.km/U.s/U.Mpc
H0_s = H0 * (1.*U.Mpc)/ ((1.*U.Mpc).to(U.km))
H0_E = (H0_s * C.hbar).to(U.GeV)
H0_E

<Quantity 1.386527729154903e-42 GeV>

In [4]:
G = 1*C.G
#Convert to energy
G_E = G* (C.hbar * C.c)**-3/(C.c**2) / (C.hbar)**-2
G_E = (G_E *(U.kg * U.m**2 / U.s**2) / U.J).to(1/U.GeV**2)
G_E

<Quantity 6.708368375728058e-39 1 / GeV2>

In [5]:
rho_c0 = 3./(8*pi*G_E) * H0_E**2
rho_c0

<Quantity 3.4207521689056005e-47 GeV4>

In [6]:
g_star = 2.
rho_m0 = 0.28 * rho_c0
rho_L0 = 0.72 * rho_c0
rho_r0 = pi**2/30 * g_star * T0_E**4
print(rho_r0)
print(rho_m0)
print(rho_L0)

1.9282126242237774e-51 GeV4
9.578106072935683e-48 GeV4
2.4629415616120323e-47 GeV4


### Problem 6.1

In [7]:
rho_k0 = np.arange(0, 3.801E-48, 1E-49) *(U.GeV**4)

In [8]:
T = 10**16*U.GeV

In [9]:
rho_c0 = rho_m0 + rho_L0 + rho_r0 + rho_k0

In [10]:
(rho_k0/rho_c0<0.1).all()

True

In [11]:
rho_k = rho_k0 * (T/T0_E)**-2
rho_m = rho_m0 * (T/T0_E)**-3
rho_r = rho_r0 * (T/T0_E)**-4
rho_c = rho_k + rho_m + rho_r+rho_L0

In [12]:
rho_k/rho_c

<Quantity [  0.00000000e+00,  2.19795659e-60,  4.39591318e-60,
             6.59386976e-60,  8.79182635e-60,  1.09897829e-59,
             1.31877395e-59,  1.53856961e-59,  1.75836527e-59,
             1.97816093e-59,  2.19795659e-59,  2.41775225e-59,
             2.63754791e-59,  2.85734356e-59,  3.07713922e-59,
             3.29693488e-59,  3.51673054e-59,  3.73652620e-59,
             3.95632186e-59,  4.17611752e-59,  4.39591318e-59,
             4.61570883e-59,  4.83550449e-59,  5.05530015e-59,
             5.27509581e-59,  5.49489147e-59,  5.71468713e-59,
             5.93448279e-59,  6.15427845e-59,  6.37407410e-59,
             6.59386976e-59,  6.81366542e-59,  7.03346108e-59,
             7.25325674e-59,  7.47305240e-59,  7.69284806e-59,
             7.91264372e-59,  8.13243938e-59,  8.35223503e-59]>

-------------------------------------
## $\frac{\rho_k(T=10^{16})}{\rho_c(T=10^{16})} < 8.35\times10^{-59}$
-------------------------------------

### Problem 6.2

Find the Hubble Length today

In [13]:
rho_tot = rho_m0 + rho_L0 + rho_r0
R_H = 1./np.sqrt(8*pi*G_E/3*rho_tot)
print(R_H)

7.21205783425972e+41 1 / GeV


Find comoving length at T = 10^16

In [14]:
T2 = 10**16 *U.GeV
a2 = T0_E/T2
d_h = a2**2 * (8*pi*G_E/3*rho_r0)**(-0.5)
print(d_h)
print(a2)
print('{:4.2e}'.format(d_h/a2))

5.2002952644234774e-14 1 / GeV
2.3266797439394582e-29
2.24e+15 1 / GeV


Find the ratio of the comoving distance today and the hubble length

In [15]:
print('ratio of comoving distance today to Hubble length today is {:2.2e}'.format((d_h/a2)/R_H))

ratio of comoving distance today to Hubble length today is 3.10e-27


## Problem 3

I find $\delta_H$ values >0.45 (larger at earlier times) during inflation, this is larger than the realistic value

## Problem 5

$t_r = 1.3\times10^{-10}$

$\rho_{\phi}(t_r) = 2.31\times10^{61}$

$a_r = 9.56\times10^{-29}$

$T(a_r)=2.4343\times10^{15}$

In [16]:
d0 = [1, 100, 1000] * U.Mpc
d0_E = (d0.to(U.m)/C.hbar/C.c).to(1/U.GeV)
print(d0_E)

[  1.56373838e+38   1.56373838e+40   1.56373838e+41] 1 / GeV


$d_r(d_0 = 1Mpc) = 1.5\times10^{10}\frac{1}{GeV}$

$d_r(d_0 = 100Mpc) = 1.5\times10^{12}\frac{1}{GeV}$

$d_r(d_0 = 1000Mpc) = 1.5\times10^{3}\frac{1}{GeV}$

## Problem 6

$\frac{\delta\rho}{\rho}(1Mpc) = 3.52\times10^{-6}$

$\frac{\delta\rho}{\rho}(100Mpc) = 3.75\times10^{-6}$

$\frac{\delta\rho}{\rho}(1000Mpc) = 3.86\times10^{-6}$

## Problem 7

In [17]:
class get_parameters(object):
    def __init__(self):
        self.rho_r_reheat = 2.3088E61*(U.GeV**4)
        self.T0 = T0_E
        self.a0 = 1.0
        self.a_reheat = 9.5594E-29
        self.T_reheat = 2.4343E15*U.GeV
        self.g_star = 2
        self.t0 = 1/H0_s
        

def calc_Temp(a, params):
    rho_r = params.rho_r_reheat * (a/params.a_reheat)**-4
    T = (rho_r/params.g_star*30/(pi**2))**0.25
    return T
    
def GeV_to_K(T_GeV):
    T_K = (T_GeV.to(U.J))*C.k_B
    return T_K

In [18]:
params = get_parameters()
tbdata = table.Table(names=['epoch','T(GeV)', 'T(K)', 'time(s)', 'time(Gyr)', 'a'], 
                     dtype=['S20', 'f', 'f', 'f', 'f', 'f'])

#### reheating

In [19]:
epoch='reheating'
a = params.a_reheat
T_GeV = params.T_reheat
T_K = GeV_to_K(T_GeV)
t = (a/params.a0)**2 * params.t0
tbdata.add_row([epoch, T_GeV, T_K, t, t.to(U.Gyr), a])

#### nucleosynthesis

In [20]:
epoch = 'nucleosynthesis'
T_n = (1.0*U.MeV).to(U.GeV)
a = params.T0/T_n * params.a0
T_GeV = calc_Temp(a, params)
T_K = GeV_to_K(T_GeV)
t = (a/params.a0)**2 * params.t0
tbdata.add_row([epoch, T_GeV, T_K, t,t.to(U.Gyr), a])

#### radiation_matter_equality

In [21]:
epoch = 'r_m_equal'
a = params.rho_r_reheat/rho_m0 * ((params.a_reheat**4)/(params.a0**3))
T_GeV = calc_Temp(a, params)
T_K = GeV_to_K(T_GeV)
t = (a/params.a0)**(1.75) * params.t0
tbdata.add_row([epoch, T_GeV, T_K, t,t.to(U.Gyr), a])

#### last scattering

In [22]:
epoch = 'last_scatter'
a = 1/1100
T_GeV = calc_Temp(a, params)
T_K = GeV_to_K(T_GeV)
t = (a/params.a0)**(1.5) * params.t0
tbdata.add_row([epoch, T_GeV, T_K, t,t.to(U.Gyr), a])

#### intercepts

In [23]:
epoch = '1Mpc_intercept'
loga = -5.7860
a = 10**loga
T_GeV = calc_Temp(a, params)
T_K = GeV_to_K(T_GeV)
t = (a/params.a0)**(1.5) * params.t0
tbdata.add_row([epoch, T_GeV, T_K, t,t.to(U.Gyr), a])

In [24]:
epoch = '100Mpc_intercept'
loga = -3.55887
a = 10**loga
T_GeV = calc_Temp(a, params)
T_K = GeV_to_K(T_GeV)
t = (a/params.a0)**(1.5) * params.t0
tbdata.add_row([epoch, T_GeV, T_K, t,t.to(U.Gyr), a])

In [25]:
epoch = '1000Mpc_intercept'
loga = -1.87271
a = 10**loga
T_GeV = calc_Temp(a, params)
T_K = GeV_to_K(T_GeV)
t = (a/params.a0)**(1.5) * params.t0
tbdata.add_row([epoch, T_GeV, T_K, t,t.to(U.Gyr), a])

#### today

In [26]:
epoch = 'today'
a = params.a0
T_GeV = calc_Temp(a, params)
T_K = GeV_to_K(T_GeV)
t = (a/params.a0)**(1.5) * params.t0
tbdata.add_row([epoch, T_GeV, T_K, t,t.to(U.Gyr), a])

In [27]:
tbdata.show_in_notebook()